pyfr/integrators/std/base.py

```diff
@@ -1,18 +1,48 @@
 # -*- coding: utf-8 -*-
 
 from pyfr.integrators.base import BaseIntegrator
+from pyfr.integrators.basecommon import BaseCommon
+from pyfr.util import proxylist
 
 
-class BaseStdIntegrator(BaseIntegrator):
+class BaseStdIntegrator(BaseCommon, BaseIntegrator):
     formulation = 'std'
 
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
+    def __init__(self, backend, systemcls, rallocs, mesh, initsoln, cfg):
 
         # Sanity checks
         if self._controller_needs_errest and not self._stepper_has_errest:
             raise TypeError('Incompatible stepper/controller combination')
 
+        # Determine the amount of temp storage required by this method
+        self.nreg = self._stepper_nregs
+
+        # Construct the relevant mesh partition
+        self.system = systemcls(backend, rallocs, mesh, initsoln,
+                                nreg=self._stepper_nregs, cfg=cfg)
+
+        super().__init__(backend, systemcls, rallocs, mesh, initsoln, cfg)
```

Comment:
Seems a bit weird passing systemcls up the chain.  After all, we've constructed the system!